modules/rtm/R/bayestools.R

```diff
@@ -0,0 +1,168 @@
+#' Generic log-likelihood generator for RTMs
+rtm_loglike <- function(nparams, model, observed, lag.max = 0.01, ...) {
+    fail_ll <- -1e10
+    stopifnot(nparams >= 1, nparams %% 1 == 0, is.function(model), is.numeric(observed))
+    n_obs <- length(observed)
+    out <- function(x) {
+        rtm_params <- x[seq_len(nparams)]
+        rsd <- x[nparams + 1]
+        mod <- model(rtm_params, ...)
+        if (any(is.na(mod))) return(fail_ll)
```

Comment:
Fail instead of ignore or na.rm?  I assume any NA is a bad sign in the LL and should failed rather than skipped?